src/Analyzers/Core/Analyzers/AbstractBuiltInCodeStyleDiagnosticAnalyzer.cs

```diff
@@ -6,10 +6,16 @@
 using System.Diagnostics;
 using System.Linq;
 using Microsoft.CodeAnalysis.Diagnostics;
-using Microsoft.CodeAnalysis.Options;
 
 namespace Microsoft.CodeAnalysis.CodeStyle
 {
+#if CODE_STYLE
+    using OptionSet = AnalyzerConfigOptions;
+    using Microsoft.CodeAnalysis.Internal.Options;
+#else
+    using Microsoft.CodeAnalysis.Options;
+#endif
```

Comment:
interesting.  why is this multiple namespaces in CODE_STYLE?

Reply:
This is an unfortunate consequence of having public types `OptionSet`, `CodeStyleOption`, `CodeStyleOptions`, etc. defined in Workspaces layer. We need these types also in the CodeStyle analyzer layer, which has no access to Workspaces. The problem comes in CodeStyle.Fixes layer, which references both CodeStyle analyzer layer and Microsoft.CodeAnalysis.Workspaces assembly (as it contains code fixes). We run into type ambiguity errors as both assemblies define the same types. So, we either need to have these types defined in a different namespace when linked into CodeStyle analyzer layer OR resort to using extern aliases to resolve ambiguity. Latter is not a viable approach as this spreads out to VB fixes and VB tests layers as VB does not even support extern aliases.